packages/angular_devkit/build_angular/src/webpack/plugins/css-optimizer.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export async function minimizeCss(
+  data: string,
+  sourceMap?: {},
+  minimizerOptions?: {
+    sourceMap?: boolean;
+  },
+) {
+  const esbuild = require('esbuild') as typeof import('esbuild');
+  const [[sourcefile, input]] = Object.entries(data);
+  const { code, warnings, map } = await esbuild.transform(input, {
+    loader: 'css',
+    minify: true,
+    sourcemap: minimizerOptions?.sourceMap && 'external',
+    sourcefile,
+  });
+
+  // Merge sourcemaps
+  let fullSourcemap;
+  if (minimizerOptions?.sourceMap) {
```

Comment:
Esbuild should be able to remap the source map if provided.  Unfortunately, it can't be passed in as an option but it will pick it up if it is in the source map url comment.
Rough code:
```
          let sourceMapLine;
          if (inputMap) {
            // esbuild will automatically remap the sourcemap if provided
            sourceMapLine = '\n//# sourceMappingURL=data:application/json;charset=utf-8;base64,';
            sourceMapLine += Buffer.from(
              typeof inputMap === 'string' ? inputMap : JSON.stringify(inputMap),
            ).toString('base64');
          }

          const { code, map, warnings } = await esbuild.transform(
            sourceMapLine ? input + sourceMapLine : input,
            {
              legalComments : 'inline',
              loader: 'css',
              minify: true,
              sourcefile,
              sourcemap: inputMap ? 'external' : false,
            },
          );
```